File: app/src/main/java/com/ed/redditapp/lib/api/RedditApiDaggerModule.java
package com.ed.redditapp.lib.api;

import com.ed.redditapp.lib.api.kotlin.KotlinRedditApi;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RedditApiDaggerModule {
    @Provides
    @Singleton
    public RedditApi provideRedditApi() {
        return new KotlinRedditApi();
    }
}

File: app/src/main/java/com/ed/redditapp/lib/api/RedditApiDaggerModule.kt
package com.ed.redditapp.lib.api

import com.ed.redditapp.lib.api.standard.StandardRedditApi
import com.ed.redditapp.lib.http.HttpClient
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RedditApiDaggerModule {
    @Provides @Singleton
    fun provideRedditApi(): RedditApi = StandardRedditApi(HttpClient())
}